kale/evaluate/uncertainty_metrics.py

```diff
@@ -0,0 +1,1036 @@
+
+                    flattened_concat_no_sep = [x for sublist in concat_no_sep for x in sublist]
+                    flattened_concat_no_sep = [x for sublist in flattened_concat_no_sep for x in sublist]
+
+                    fold_all_bins_concat_lms_nosep[idx_bin] = (
+                        fold_all_bins_concat_lms_nosep[idx_bin] + flattened_concat_no_sep
+                    )
+
+                    for lm in range(len(landmarks)):
+                        fold_all_bins_concat_lms_sep_foldwise[lm][idx_bin] = (
+                            fold_all_bins_concat_lms_sep_foldwise[lm][idx_bin]
+                            + return_dict["all bins concatenated lms seperated"][lm][idx_bin]
+                        )
+
+                        if return_dict["all bins concatenated lms seperated"][lm][idx_bin] != []:
+                            combined = (
+                                fold_all_bins_concat_lms_sep_all[lm][idx_bin]
+                                + return_dict["all bins concatenated lms seperated"][lm][idx_bin][0]
+                            )
+                        else:
+                            combined = fold_all_bins_concat_lms_sep_all[lm][idx_bin]
+
+                        fold_all_bins_concat_lms_sep_all[lm][idx_bin] = combined
+
+            # reverse orderings
+            fold_mean_bins = fold_mean_bins[::-1]
+            fold_all_bins = fold_all_bins[::-1]
+            fold_all_bins_concat_lms_nosep = fold_all_bins_concat_lms_nosep[::-1]
+            fold_all_bins_concat_lms_sep_foldwise = [x[::-1] for x in fold_all_bins_concat_lms_sep_foldwise]
+            fold_all_bins_concat_lms_sep_all = [x[::-1] for x in fold_all_bins_concat_lms_sep_all]
+
+            all_mean_error_bins[model + " " + uncertainty_type] = fold_mean_bins
+            all_mean_error_bins_lms_sep[model + " " + uncertainty_type] = fold_all_bins
+
+            all_concat_error_bins_lm_nosep[model + " " + uncertainty_type] = fold_all_bins_concat_lms_nosep
+
+            for lm in range(len(fold_all_bins_concat_lms_sep_foldwise)):
+                all_concat_error_bins_lm_sep_foldwise[lm][
+                    model + " " + uncertainty_type
+                ] = fold_all_bins_concat_lms_sep_foldwise[lm]
+                all_concat_error_bins_lm_sep_all[lm][model + " " + uncertainty_type] = fold_all_bins_concat_lms_sep_all[
+                    lm
+                ]
+
+    return {
+        "all mean error bins nosep": all_mean_error_bins,
+        "all mean error bins lms sep": all_mean_error_bins_lms_sep,
+        "all error concat bins lms nosep": all_concat_error_bins_lm_nosep,
+        "all error concat bins lms sep foldwise": all_concat_error_bins_lm_sep_foldwise,
+        "all error concat bins lms sep all": all_concat_error_bins_lm_sep_all,
+    }
+
+
+def evaluate_jaccard(bin_predictions, uncertainty_pairs, num_bins, landmarks, num_folds=8, combine_middle_bins=False):
```

Comment:
Sure no existing API for Jaccard eleswhere?

Reply:
I checked and it exists on sklean [here](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.jaccard_score.html). i implemented myself in 3 lines in jaccard_similarity(). is it worth removing and changing to their API?

Reply:
in discussion: https://github.com/pykale/pykale/discussions/392